### PR TITLE
CoreVideo - deepsleep when off

### PR DESCRIPTION
**Pull request by ladislas**
related story/pr: #1101

### CODE REVIEW

File: tests/functional/tests/deep_sleep_core_video/suite_core_video.cpp [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+			when("I turn off") = [] {
+				corevideo.turnOff();
+				rtos::ThisThread::sleep_for(250ms);
+
+				then("I expect deep sleep TO BE possible") = [] {
+					auto status = utils::sleep::system_deep_sleep_check();
+
+					expect(status.can_deep_sleep);
+					expect(status.test_check_ok);
```
[the remark this answers is not in the document]
Comment (YannLocatelli) [reply tests/functional/tests/deep_sleep_core_video/suite_core_video.cpp:98]: I do not have explanation for now, investigating for now~

**Review by YannLocatelli — CHANGES_REQUESTED** [review]
Thank you for your work~

Still a bug for functional test only and some other comments

File: tests/functional/tests/deep_sleep_core_video/suite_core_video.cpp [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+			when("I do nothing") = [] {
+				then("I expect deep sleep TO BE possible") = [] {
+					auto status = utils::sleep::system_deep_sleep_check();
+
+					expect(status.can_deep_sleep);
+					expect(status.test_check_ok);
+				};
+			};
+		};
+	};
```
[the remark this answers is not in the document]
Comment (YannLocatelli) [reply tests/functional/tests/deep_sleep_core_video/suite_core_video.cpp:58]: There is a lot of initialization in this file, not sure it helps

**Comment by ladislas** [comment]
replaced by #1173

**Pull request closed without merging**